Travis CI fails on MacOS due to QT install issue

Travis CI fails on MacOS due to QT install issue. Perhaps due to changes to Brew?

**Error:**
```
CMake Error at /usr/local/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:15 (message):
  The imported target "Qt5::Core" references the file
     "/usr/local/.//mkspecs/macx-clang"
  but this file does not exist.  Possible reasons include:
  * The file was deleted, renamed, or moved to another location.
  * An install or uninstall procedure did not complete successfully.
  * The installation package was faulty and contained
     "/usr/local/lib/cmake/Qt5Core/Qt5CoreConfigExtras.cmake"
  but not all the files it references.
Call Stack (most recent call first):
  /usr/local/lib/cmake/Qt5Core/Qt5CoreConfigExtras.cmake:50 (_qt5_Core_check_file_exists)
  /usr/local/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:166 (include)
  /usr/local/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package)
  CMakeLists.txt:82 (find_package)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -93,6 +93,7 @@ install:
       QT5_CMAKE_PREFIX=$(ls -d $(brew --cellar)/qt/*/lib/cmake)
       CMAKE_OPTS="-DCMAKE_PREFIX_PATH=${QT5_CMAKE_PREFIX}"
     fi
+    export CMAKE_OPTS
   
 script:
   - mkdir build

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,7 +17,8 @@ env:
     - CCACHE_TEMPDIR=/tmp/.ccache-temp
     - CCACHE_COMPRESS=1
     - CASHER_TIME_OUT=1000
-
+    - QT=510 BREW=@5.10
+    
 addons_shortcuts:
   addons_clang50: &clang50
     apt:
@@ -86,10 +87,12 @@ install:
     if [ "$TRAVIS_OS_NAME" = "linux" ]; then
       sudo apt-get install -qq qt59base qt59declarative qt59quickcontrols2; source /opt/qt59/bin/qt59-env.sh;
     else
-      brew install qt;
-      brew link --force qt && sudo ln -s /usr/local/Cellar/qt/5.10.1/mkspecs /usr/local/mkspecs && sudo ln -s /usr/local/Cellar/qt/5.10.1/plugins /usr/local/plugins;
+      brew install cppcheck qt$BREW;
+      bbrew link --force qt$BREW;
     fi
-
+before_script:
+  - '[[ "$TRAVIS_OS_NAME" != linux ]] || . /opt/qt$QT/bin/qt$QT-env.sh'
+  
 script:
   - mkdir build
   - cd build

--- a/.travis.yml
+++ b/.travis.yml
@@ -91,7 +91,7 @@ install:
       brew link --force qt$BREW;
     fi
 before_script:
-  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then . /opt/qt$QT/bin/qt$QT-env.sh; fi
+  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then /opt/qt$QT/bin/qt$QT-env.sh; fi
   
 script:
   - mkdir build

--- a/.travis.yml
+++ b/.travis.yml
@@ -86,16 +86,17 @@ install:
   - |
     if [ "$TRAVIS_OS_NAME" = "linux" ]; then
       sudo apt-get install -qq qt59base qt59declarative qt59quickcontrols2; source /opt/qt59/bin/qt59-env.sh;
+      CMAKE_OPTS=""
     else
       brew install cppcheck qt;
       brew link --force qt;
+      QT5_CMAKE_PREFIX=$(ls -d $(brew --cellar)/qt/*/lib/cmake)
+      CMAKE_OPTS="-DCMAKE_PREFIX_PATH=${QT5_CMAKE_PREFIX}"
     fi
-before_script:
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then /opt/qt$QT/bin/qt$QT-env.sh; fi
   
 script:
   - mkdir build
   - cd build
-  - cmake .. -DCMAKE_BUILD_TYPE=${BUILD} -DCMAKE_CXX_FLAGS="${CXX_FLAGS}" -DCMAKE_EXE_LINKER_FLAGS="${CXX_LINKER_FLAGS}" -DBUILD_WITH_CCACHE=1  -DCLANG_TIDY=$(which ${CLANG_TIDY})
+  - cmake .. -DCMAKE_BUILD_TYPE=${BUILD} ${CMAKE_OPTS} -DCMAKE_CXX_FLAGS="${CXX_FLAGS}" -DCMAKE_EXE_LINKER_FLAGS="${CXX_LINKER_FLAGS}" -DBUILD_WITH_CCACHE=1  -DCLANG_TIDY=$(which ${CLANG_TIDY})
   - make VERBOSE=1
 # ctest --output-on-failure

--- a/.travis.yml
+++ b/.travis.yml
@@ -86,14 +86,14 @@ install:
   - |
     if [ "$TRAVIS_OS_NAME" = "linux" ]; then
       sudo apt-get install -qq qt59base qt59declarative qt59quickcontrols2; source /opt/qt59/bin/qt59-env.sh;
-      CMAKE_OPTS=""
+      CMAKE_OPTS="";
     else
       brew install cppcheck qt;
       brew link --force qt;
-      QT5_CMAKE_PREFIX=$(ls -d $(brew --cellar)/qt/*/lib/cmake)
-      CMAKE_OPTS="-DCMAKE_PREFIX_PATH=${QT5_CMAKE_PREFIX}"
+      QT5_CMAKE_PREFIX=$(ls -d $(brew --cellar)/qt/*/lib/cmake);
+      CMAKE_OPTS="-DCMAKE_PREFIX_PATH=${QT5_CMAKE_PREFIX}";
     fi
-    export CMAKE_OPTS
+    export CMAKE_OPTS;
   
 script:
   - mkdir build

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,7 +17,7 @@ env:
     - CCACHE_TEMPDIR=/tmp/.ccache-temp
     - CCACHE_COMPRESS=1
     - CASHER_TIME_OUT=1000
-    - QT=510 BREW=@5.10
+    - QT=511
     
 addons_shortcuts:
   addons_clang50: &clang50
@@ -87,8 +87,8 @@ install:
     if [ "$TRAVIS_OS_NAME" = "linux" ]; then
       sudo apt-get install -qq qt59base qt59declarative qt59quickcontrols2; source /opt/qt59/bin/qt59-env.sh;
     else
-      brew install cppcheck qt$BREW;
-      brew link --force qt$BREW;
+      brew install cppcheck qt;
+      brew link --force qt;
     fi
 before_script:
   - if [ "$TRAVIS_OS_NAME" = "osx" ]; then /opt/qt$QT/bin/qt$QT-env.sh; fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -86,8 +86,8 @@ install:
     if [ "$TRAVIS_OS_NAME" = "linux" ]; then
       sudo apt-get install -qq qt59base qt59declarative qt59quickcontrols2; source /opt/qt59/bin/qt59-env.sh;
     else
-      brew install qt5;
-      brew link --force qt5 && sudo ln -s /usr/local/Cellar/qt/5.10.1/mkspecs /usr/local/mkspecs && sudo ln -s /usr/local/Cellar/qt/5.10.1/plugins /usr/local/plugins;
+      brew install qt;
+      brew link --force qt && sudo ln -s /usr/local/Cellar/qt/5.10.1/mkspecs /usr/local/mkspecs && sudo ln -s /usr/local/Cellar/qt/5.10.1/plugins /usr/local/plugins;
     fi
 
 script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -98,6 +98,11 @@ install:
 script:
   - mkdir build
   - cd build
-  - cmake .. -DCMAKE_BUILD_TYPE=${BUILD} ${CMAKE_OPTS} -DCMAKE_CXX_FLAGS="${CXX_FLAGS}" -DCMAKE_EXE_LINKER_FLAGS="${CXX_LINKER_FLAGS}" -DBUILD_WITH_CCACHE=1  -DCLANG_TIDY=$(which ${CLANG_TIDY})
+  - |
+    if [ "$TRAVIS_OS_NAME" = "osx" ]; then
+      cmake .. -DCMAKE_BUILD_TYPE=${BUILD} -DCMAKE_PREFIX_PATH="$(ls -d $(brew --cellar)/qt/*/lib/cmake)" -DCMAKE_CXX_FLAGS="${CXX_FLAGS}" -DCMAKE_EXE_LINKER_FLAGS="${CXX_LINKER_FLAGS}" -DBUILD_WITH_CCACHE=1  -DCLANG_TIDY=$(which ${CLANG_TIDY});
+    else
+      cmake .. -DCMAKE_BUILD_TYPE=${BUILD} -DCMAKE_CXX_FLAGS="${CXX_FLAGS}" -DCMAKE_EXE_LINKER_FLAGS="${CXX_LINKER_FLAGS}" -DBUILD_WITH_CCACHE=1  -DCLANG_TIDY=$(which ${CLANG_TIDY});
+    fi
   - make VERBOSE=1
 # ctest --output-on-failure

--- a/.travis.yml
+++ b/.travis.yml
@@ -88,10 +88,10 @@ install:
       sudo apt-get install -qq qt59base qt59declarative qt59quickcontrols2; source /opt/qt59/bin/qt59-env.sh;
     else
       brew install cppcheck qt$BREW;
-      bbrew link --force qt$BREW;
+      brew link --force qt$BREW;
     fi
 before_script:
-  - '[[ "$TRAVIS_OS_NAME" != linux ]] || . /opt/qt$QT/bin/qt$QT-env.sh'
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then . /opt/qt$QT/bin/qt$QT-env.sh; fi
   
 script:
   - mkdir build

